application/src/main/java/org/eclipse/hono/application/ApplicationConfig.java

```diff
@@ -36,20 +38,36 @@
 @Configuration
 public class ApplicationConfig {
 
+    private final VertxOptions vertxOptions;
+
     /**
-     * Gets the singleton Vert.x instance to be used by Hono.
-     * 
-     * @return the instance.
+     * Adds metrics options to the Vert.x configuration, to write the Vert.x metrics to the configured Hono metrics server
+     *
+     * @param metricsOptions
      */
-    @Bean
-    public Vertx vertx() {
-        VertxOptions options = new VertxOptions()
+    @Autowired(required = false)
+    public ApplicationConfig(final MetricsOptions metricsOptions) {
```

Comment:
We could do that, but then we need to autowire it in all usages. I would prefer to autowire the MetricsOptions in the constructor, but remember than only the MetricsOptions and build die VertexOptions as before and add the MetricsOptions if they are unqueal null. The pattern autowire and remember a bean in constructor is also mentioned directly in the description of the Configuration annotation. What do you think?